src/main/java/com/robosh/ejournal/service/AdminService.java

```diff
@@ -1,8 +1,31 @@
 package com.robosh.ejournal.service;
 
+import com.robosh.ejournal.data.dto.admin.AdminInfoDto;
+import com.robosh.ejournal.data.dto.admin.UpdateAdminDto;
+import com.robosh.ejournal.data.entity.admin.Admin;
+import com.robosh.ejournal.data.mapping.admin.AdminMapper;
+import com.robosh.ejournal.data.repository.AdminRepository;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
+import javax.validation.ValidationException;
+
 @Service
+@RequiredArgsConstructor
 public class AdminService {
 
+    private final AdminRepository adminRepository;
+    private final AdminMapper adminMapper;
+
+    public AdminInfoDto save(UpdateAdminDto updateAdminDto) {
+        if (!updateAdminDto.getPassword().equals(updateAdminDto.getConfirmedPassword())) {
+            throw new ValidationException("Password should be same");
+        }
+        Admin admin = adminMapper.fromUpdateAdminDtoToAdmin(updateAdminDto);
+        if (updateAdminDto.getId() == null) {
```

Comment:
why do you get id from updateAdminDto but not from admin variable?

Reply:
because if id in dto is null, we should set school to null if don't do it, we will have a school with all null values, in another word, a school will create via constructor without parameters